mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -784,10 +784,13 @@ public void onSuccess(Call call, int statusCode, Headers headers, byte[] respons
                 mPendingCall = null;
                 mInitState = InitState.DONE;
 
+                String defaultSitemapName =
+                        mSettings.getString(Constants.PREFERENCE_SITEMAP_NAME, "");
+
                 // OH1 returns XML, later versions return JSON
                 List<OpenHABSitemap> result = mOpenHABVersion == 1
-                        ? loadSitemapsFromXml(responseBody)
-                        : loadSitemapsFromJson(responseBody);
+                        ? loadSitemapsFromXml(responseBody, defaultSitemapName)
+                        : loadSitemapsFromJson(responseBody, defaultSitemapName);
                 Log.d(TAG, "Server returned sitemaps: " + result);
                 mSitemapList.clear();
                 if (result != null) {
```

Comment:
... sort `result` here (or `mSitemapList` below).

Separating those also makes for better unit testing.